internal/reconcile/service_monitor.go

```diff
@@ -0,0 +1,36 @@
+package reconcile
+
+import (
+	"context"
+	"fmt"
+	log "github.com/ViaQ/logerr/v2/log/static"
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/openshift/cluster-logging-operator/internal/utils/comparators/servicemonitor"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//ServiceMonitor reconciles a ServiceMonitor to the desired spec returning an error
+//if there is an issue creating or updating to the desired state
+func ServiceMonitor(k8Client client.Client, desired *monitoringv1.ServiceMonitor) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		current := &monitoringv1.ServiceMonitor{}
+		key := client.ObjectKeyFromObject(desired)
+		if err := k8Client.Get(context.TODO(), key, current); err != nil {
+			if errors.IsNotFound(err) {
+				return k8Client.Create(context.TODO(), desired)
+			}
+			return fmt.Errorf("failed to get %v servicemonitor: %v", key, err)
```

Comment:
Use `%w` if you want to use the "error embedding" of fmt:

```suggestion
			return fmt.Errorf("failed to get %v servicemonitor: %w", key, err)
```